resources/wikia/libraries/modil/modil.js

```diff
@@ -134,6 +138,17 @@
 	 * @throws {Error} If dependenices is not undefined but not an array
 	 */
 	define = function (id, dependencies, definition, defMock) {
+		//no id, it's the definition or dependencies
+		if (id instanceof funcType && !dependencies && !definition) {
+			definition = id;
+			dependencies = nil;
+			id = getRandomId();
+		} else if (id instanceof arrType && dependencies && !definition) {
+			definition = dependencies;
+			dependencies = id;
+			id = getRandomId();
```

Comment:
It seems we'll have `id` defined almost all the time now, so changing JSDocs and the error message below may be helpful. What do you think?